storage/innobase/row/row0ins.cc

```diff
@@ -1279,9 +1279,8 @@ row_ins_foreign_check_on_constraint(
 						 clust_index, tmp_heap);
 	}
 
-	if (cascade->is_delete && foreign->v_cols != NULL
-	    && foreign->v_cols->size() > 0
-	    && table->vc_templ == NULL) {
+	if (cascade->is_delete && dict_table_get_n_v_cols(table) > 0
+			&& table->vc_templ == NULL) {
```

Comment:
Can you please add a comment on why we initialize virtual column template?

Some thing like
```
/* A cascade delete from the parent table triggers delete on the child table. Before a clustered index record is deleted in the child table, a copy of row is built to remove secondary index records. This copy of the row requires virtual columns to be materialized. Hence, if child table has any virtual columns, we have to initialize virtual column template */
```
Please lookout for 80 column char limit when pasting the comment to code :wink:

Reply:
thank for well explained comment. Added to the code.

Reply:
@rahulmalik87 Please consider fixing commit message too :)

Reply:
Let me know if commit message is fine now.

Reply:
This && should start position below 'c' in cascade in above line

Reply:
fixed